vnext/Microsoft.ReactNative.Cxx/JSValueXaml.h

```diff
@@ -19,6 +19,53 @@ inline void ReadValue(JSValue const &jsValue, Windows::UI::Color &value) noexcep
   value = XamlHelper::ColorFrom([&jsValue](IJSValueWriter const &writer) noexcept { jsValue.WriteTo(writer); });
 }
 
+inline void ReadValue(JSValue const &jsValue, xaml::Thickness &value) noexcept {
+  if (auto array = jsValue.TryGetArray()) {
+    if (array->size() == 4) {
+      value = xaml::ThicknessHelper::FromLengths(
+          (*array)[0].AsDouble(), (*array)[1].AsDouble(), (*array)[2].AsDouble(), (*array)[3].AsDouble());
+      return;
+    }
+  } else if (auto number = jsValue.TryGetDouble()) {
+    value = xaml::ThicknessHelper::FromUniformLength(*number);
+  } else if (auto numberInt = jsValue.TryGetInt64()) {
+    const auto valueDbl = static_cast<double>(*numberInt);
+    value = xaml::ThicknessHelper::FromUniformLength(valueDbl);
+  } else {
+    const auto &obj = jsValue.AsObject();
+    value = xaml::ThicknessHelper::FromLengths(
+        obj["left"].AsDouble(), obj["top"].AsDouble(), obj["right"].AsDouble(), obj["bottom"].AsDouble());
+    return;
+  }
+}
+
+inline void ReadValue(JSValue const &jsValue, xaml::CornerRadius &value) noexcept {
+  if (auto array = jsValue.TryGetArray()) {
+    if (array->size() == 4) {
+      value = xaml::CornerRadiusHelper::FromRadii(
+          (*array)[0].AsDouble(), (*array)[1].AsDouble(), (*array)[2].AsDouble(), (*array)[3].AsDouble());
+      return;
+    }
+  } else if (auto number = jsValue.TryGetDouble()) {
+    value = xaml::CornerRadiusHelper::FromUniformRadius(*number);
+  } else if (auto numberInt = jsValue.TryGetInt64()) {
+    const auto valueDbl = static_cast<double>(*numberInt);
+    value = xaml::CornerRadiusHelper::FromUniformRadius(valueDbl);
+  } else {
+    const auto &obj = jsValue.AsObject();
+    value = xaml::CornerRadiusHelper::FromRadii(
+        obj["topLeft"].AsDouble(),
+        obj["topRight"].AsDouble(),
+        obj["bottomRight"].AsDouble(),
+        obj["bottomLeft"].AsDouble());
+    return;
+  }
+}
+
+inline void ReadValue(JSValue const &jsValue, winrt::Windows::Foundation::Uri &value) noexcept {
+  value = Uri{winrt::to_hstring(jsValue.AsString())};
```

Comment:
correct, it will crash. There isn't anything reasonable to do, either crash, or continue with an empty / default Uri, and probably crash later. Crashing as soon as possible is safest.

Reply:
I remember a while ago you added something to make crashes due to thread affinity related issues more visible. I think the direct feedback that it provides is helpful for people without C++ debugging experience.

Is it possible to provide user feedback in any other generic cases where we terminate.